torchci/components/JobAnnotationToggle.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import { JobData } from "../lib/types";
+import { ToggleButtonGroup, ToggleButton } from "@mui/material";
+
+type JobAnnotation =
+  | "infra_flake"
+  | "time_out"
+  | "SEV"
+  | "broken_trunk"
+  | "test_flake"
+  | "null";
+
+const testFailureRe = /^(?:FAIL|ERROR) \[.*\]: (test_.* \(.*Test.*\))/;
+
+export default function JobAnnotationToggle({ job }: { job: JobData }) {
+  const [state, setState] = React.useState<JobAnnotation>(
+    (job.failureAnnotation ?? "null") as JobAnnotation
+  );
+
+  async function handleChange(
+    _: React.MouseEvent<HTMLElement>,
+    newState: JobAnnotation
+  ) {
+    setState(newState);
```

Comment:
We should prob use a json object as the newState instead of a string, so that we can add extra annotations in the future (such as a freeform text area)

Reply:
The actual saving gets done on the backend so you can just add a new state here or something for the free form text or changing it to a JSON. I think I'll skip this since any changes to this would require some UI work and other things anyways to make it support free form text.